### PR TITLE
Add MSBuild support for controlling --verbosity, --configuration, --platform and project file arguments to slngen.exe

### PR DESCRIPTION
**Pull request by jdrobison**
This change gives project authors greater control over the parameters passed to `slngen.exe` by the `SlnGen` task:

| `slngen.exe` Parameter | MSBuild property/item | Default value |
|----|----|----|
| `--verbosity` | `SlnGenVerbosity` (property) | `Normal` |
| `--configuration` | `SlnGenConfiguration` (property) | Unspecified |
| `--platform` | `SlnGenPlatform` (property) | Unspecified  |
| Projects in solution | `SlnGenProjects` (item) | The project being built |

Here's an example of project authoring using the new MSBuild properties and items:

```xml
<PropertyGroup Label="SlnGen configuration properties">
  <SlnGenLaunchVisualStudio>false</SlnGenLaunchVisualStudio>
  <SlnGenSolutionFileFullPath>$(MSBuildProjectDirectory)\$(MSBuildProjectName).sln</SlnGenSolutionFileFullPath>
  <SlnGenFolders>true</SlnGenFolders>
  <SlnGenVerbosity>Minimal</SlnGenVerbosity>
  <SlnGenPlatform>AnyCPU</SlnGenPlatform>
  <SlnGenConfiguration>Debug;Release</SlnGenConfiguration>
</PropertyGroup>
<ItemGroup Label="SlnGen configuration items">
  <SlnGenProjects Include="**\*.csproj" />
  <SlnGenProjects Include="..\..\..\UnitTests\**\File*.csproj" />
</ItemGroup>
```

### CODE REVIEW

**Comment by jdrobison** [comment]
Does a completed PR automatically result in publishing an updated NuGet package, or is there a separate process for that?  If the former, what is the typical latency between PR completion and the NuGet package update?

**Comment by mruxmohan4** [comment]
There is a separate process which requires approval for package updates; however, there should be minimal latency between PR completion and package updates (should be in effect same day as PR completion).

**Comment by jdrobison** [comment]
> There is a separate process which requires approval for package updates; however, there should be minimal latency between PR completion and package updates (should be in effect same day as PR completion).

Ah, is that the process that's documented in [RELEASE.md](https://github.com/microsoft/slngen/blob/main/RELEASE.md)?

**Comment by mruxmohan4** [comment]
Yes, I will add a note that repo maintainers are responsible for those package pushes. 
Just reviewed the issue and changes - MSBuild properties/items seem appropriate for this use case.
Thank you for your contribution.

Regarding https://microsoft.github.io/slngen/, I will create a separate PR to update documentation (along with the changes made here). Generally, those changes are made here: https://github.com/microsoft/slngen/tree/main/docs

**Review by mruxmohan4 — APPROVED** [review]

**Pull request merged**
